src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogView.java

```diff
@@ -0,0 +1,120 @@
+    @FXML private Button btnBrowse;
+    @FXML private CheckBox chkAutosave;
+    @FXML private ButtonType connectButton;
+
+    @Inject private DialogService dialogService;
+
+    private final Button btnConnect;
+
+    private SharedDatabaseLoginDialogViewModel viewModel;
+    private final ControlsFxVisualizer visualizer = new ControlsFxVisualizer();
+
+    public SharedDatabaseLoginDialogView(JabRefFrame frame) {
+        this.frame = frame;
+        this.setTitle(Localization.lang("Connect to shared database"));
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        ControlHelper.setAction(connectButton, this.getDialogPane(), event -> openDatabase());
+        btnConnect = (Button) this.getDialogPane().lookupButton(connectButton);
+        setLoadingConnectButtonText();
+        btnConnect.disableProperty().bind(viewModel.formValidation().validProperty().not());
+
+    }
+
+    @FXML
+    private void openDatabase() {
+        viewModel.openDatabase();
+    }
+
+    @FXML
+    private void initialize() {
+        visualizer.setDecoration(new IconValidationDecorator());
+
+        viewModel = new SharedDatabaseLoginDialogViewModel(frame, dialogService);
+        cmbDbType.itemsProperty().bind(viewModel.dbmstypeProperty());
+        cmbDbType.getSelectionModel().select(0);
+
+        tbDb.textProperty().bindBidirectional(viewModel.dbProperty());
+        tbHost.textProperty().bindBidirectional(viewModel.hostProperty());
+        tbUser.textProperty().bindBidirectional(viewModel.userProperty());
+        tbPwd.textProperty().bindBidirectional(viewModel.passwordProperty());
+        tbPort.textProperty().bindBidirectional(viewModel.portProperty());
+        cmbDbType.valueProperty().bindBidirectional(viewModel.selectedDbmstypeProperty());
+
+        tbFolder.textProperty().bindBidirectional(viewModel.folderProperty());
+        btnBrowse.disableProperty().bind(chkAutosave.selectedProperty().not());
+        tbFolder.disableProperty().bind(chkAutosave.selectedProperty().not());
+        chkAutosave.selectedProperty().bindBidirectional(viewModel.autosaveProperty());
+
+        Platform.runLater(() -> {
+            visualizer.initVisualization(viewModel.dbValidation(), tbDb, true);
+            visualizer.initVisualization(viewModel.hostValidation(), tbHost, true);
+            visualizer.initVisualization(viewModel.portValidation(), tbPort, true);
+            visualizer.initVisualization(viewModel.userValidation(), tbUser, true);
+
+            EasyBind.subscribe(chkAutosave.selectedProperty(), selected -> {
+                visualizer.initVisualization(viewModel.folderValidation(), tbFolder, true);
+            });
+        });
+
+        viewModel.applyPreferences();
+
+    }
+
+    @FXML
+    void openFileDialog(ActionEvent event) {
+        viewModel.openFileDialog();
+    }
+
+    private void setLoadingConnectButtonText() {
+
+        if (viewModel.loadingProperty().get()) {
```

Comment:
This should be done via binding, because otherwise setting `loading = true` does not change the text to "Connecting...". You can use `EasyBind.map` to convert the boolean to the appropriate text and then bind this text to `btnConnect.textProperty`